In [0]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import nltk
nltk.download('stopwords')
nltk.download('wordnet')

In [0]:
data_inter = pd.read_excel('Twitter EN Interactions.xlsx')
data_post = pd.read_excel('Twitter EN Posts.xlsx')
data_inter.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 129838 entries, 0 to 129837
Data columns (total 14 columns):
No                129838 non-null int64
Date              129838 non-null object
Author            129838 non-null object
Post              129838 non-null object
Original Reach    129838 non-null int64
Viral Reach       129838 non-null int64
Engagements       129838 non-null int64
Viral Score       129821 non-null float64
Sentiment         129838 non-null object
Link              129838 non-null object
Location          129838 non-null object
Mood              0 non-null float64
Sentence Type     0 non-null float64
Tags              0 non-null float64
dtypes: float64(4), int64(4), object(6)
memory usage: 13.9+ MB


In [0]:
data_post.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11477 entries, 0 to 11476
Data columns (total 14 columns):
No                11477 non-null int64
Date              11477 non-null object
Author            11477 non-null object
Post              11477 non-null object
Original Reach    11477 non-null int64
Viral Reach       11477 non-null int64
Engagements       11477 non-null int64
Viral Score       11473 non-null float64
Sentiment         11477 non-null object
Link              11477 non-null object
Location          11477 non-null object
Mood              0 non-null float64
Sentence Type     0 non-null float64
Tags              0 non-null float64
dtypes: float64(4), int64(4), object(6)
memory usage: 1.2+ MB


In [0]:
posts_inter = data_inter['Post']
posts_post = data_post['Post']
print('Data Twitter EN Interactions : ',len(posts_inter))
print('Data Twitter EN Posts : ', len(posts_post))


Data Twitter EN Interactions :  129838
Data Twitter EN Posts :  11477


## Cleaning without stemming and stopwords removal

In [0]:
from bs4 import BeautifulSoup
import re
from nltk.tokenize import WordPunctTokenizer
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer

wordnet_lemmatizer = WordNetLemmatizer()
tok = WordPunctTokenizer()
pat1 = r'@[A-Za-z0-9_]+' #remove mention
pat2 = r'https?://[A-Za-z0-9./]+' #remove link
pat3 = r'RT' #remove RT
pat4 = r'#[A-Za-z0-9_]+' #remove hashtag
pat5 = r'pic.twitter.com/[A-Za-z0-9]+'
combined_pat = r'|'.join((pat1,pat2,pat3,pat4,pat5))
stop_words = set(stopwords.words('english')) 

def tweet_cleaner(text):
    soup = BeautifulSoup(text, 'lxml')
    souped = soup.get_text()
    stripped = re.sub(combined_pat, '', souped) #delete pattern combined_pat in souped
    try:
        clean = stripped.decode("utf-8").replace(u"\ufffd", "?")
    except:
        clean = stripped
    letters_only = re.sub("[^a-zA-Z]", " ", clean)
    lower_case = letters_only.lower()
    #words = tok.tokenize(lower_case)
    #lemmatized = [wordnet_lemmatizer.lemmatize(word) for word in words]
    #filtered_words = [w for w in lemmatized if not w in stop_words] 
    #return (" ".join(filtered_words)).strip()
    return lower_case

In [0]:
from tqdm import tqdm
import time

with tqdm(total=len(posts_inter)) as pbar:
  clean_tweets_inter = []
  for i in range(len(posts_inter)):
    clean_tweets_inter.append(tweet_cleaner(posts_inter[i]))
    time.sleep(0.1)
    pbar.update(10)
  
clean_tweets_post = [tweet_cleaner(posts_post[i]) for i in range(len(posts_post))]

786100it [2:13:59, 97.81it/s]

Buffered data was truncated after reaching the output size limit.

In [0]:
clean_tweets_inter[:20]

['  yes sir but don t make fake news  be careful ',
 '   protesters burn local parliament building in west papua ',
 '   as vero notes below  this is the largest protest in jayapura in a long time  this might get serious              jayapura  west papua  ribuan massa teriakkan  papua merdeka  dalam aksi menentang rasisme   thousands of people shouting  free west papua  in the rally against racism  this is the biggest protest in years  ',
 '   this year      at least     people  mostly women and children  died  some at the hands of the security forces  many others died of hunger or sickness while fleeing their conflict ridden villages      today s minkes  racism at heart of jakarta papua conflict   ',
 '   indonesia s jokowi urges calm after violent west papua protests ',
 '   protesters burn local parliament building in west papua ',
 'admin  rasa buzzer hoax    saya meminta klarifikasi dan permintaan maaf terbuka dari  atas pencemaran nama baik terhadap saya   ini negara hukum  saya 

In [0]:
clean_tweets_post[:20]

[' wp     wiranto discloses conspiracy involving separatist leader benny wenda   joe collins  ',
 'unrest has been reported on the streets of manokwari after reports of racist abuse of papuan students          ',
 'indonesian president calls for calm after violent protests in west papua ',
 'indonesian president calls for calm after violent protests in west papua ',
 'dani tribe chief eli mabel is pictured holding the remains of agat mamete mabel in the village of wogi in wamena in west papua  an island in the centre of papua new guinea   the indigenous tribe  who live in a    ',
 'parliament set ablaze in indonesia unrest  ',
 'protesters burn local parliament building in west papua ',
 'the neighbourhood ',
 '  originally  papua doesn t belong to indonesia  our anchestor voiced our independence but still fought to claimed papua as our region   convenants  conferences  even armed combat  their efforts is not a thing to teach us for being a racist people ',
 'thousands riot in papua  p

In [0]:
print("Jumlah Tweet EN Interaction : ",len(clean_tweets_inter))
clean_tweets_remove_dups = list(dict.fromkeys(clean_tweets_inter))
print("Jumlah Tweet EN Interaction tanpa duplikat : ", len(clean_tweets_remove_dups))

Jumlah Tweet ID Interaction :  129838
Jumlah Tweet ID Interaction tanpa duplikat :  17707


In [0]:
print("Jumlah Tweet EN Post : ",len(clean_tweets_post))
clean_tweets_remove_dups2 = list(dict.fromkeys(clean_tweets_post))
print("Jumlah Tweet EN Post tanpa duplikat : ", len(clean_tweets_remove_dups2))

Jumlah Tweet EN Post :  11477
Jumlah Tweet EN Post tanpa duplikat :  8215


In [0]:
129838+11477

## DATA TWEET BAHASA INGGRIS

- Jumlah tweet EN interactions : 129.838
- Jumlah tweet tanpa duplikat  : 17.707



- Jumlah tweet EN posts       : 11.477
- Jumlah tweet tanpa duplikat : 8215



- Total tweet EN             : 141.315
- Total tweet tanpa duplikat : 25.922

## Stemming and Stopwords Removal

In [0]:
from nltk.tokenize import WordPunctTokenizer
from nltk.corpus import stopwords
from nltk.tokenize import WordPunctTokenizer
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer

tok = WordPunctTokenizer()
def tweet_cleaner2(text):
  
    words = tok.tokenize(text)
    lemmatized = [wordnet_lemmatizer.lemmatize(word) for word in words]
    filtered_words = [w for w in lemmatized if not w in stop_words] 
    return (" ".join(filtered_words)).strip()
 
 

In [0]:
from tqdm import tqdm
import time

with tqdm(total=len(clean_tweets_inter)) as pbar:
  clean_tweets_inter2 = []
  for i in range(len(clean_tweets_inter)):
    clean_tweets_inter2.append(tweet_cleaner2(clean_tweets_inter[i]))
    pbar.update()
  
clean_tweets_post2 = [tweet_cleaner2(clean_tweets_post[i]) for i in range(len(clean_tweets_post))]

100%|██████████| 129838/129838 [00:18<00:00, 7198.37it/s]


In [0]:
with tqdm(total=len(clean_tweets_remove_dups)) as pbar:
  clean_tweets_inter_remove_dups = []
  for i in range(len(clean_tweets_remove_dups)):
    clean_tweets_inter_remove_dups.append(tweet_cleaner2(clean_tweets_remove_dups[i]))
    time.sleep(0.1)
    pbar.update(10)
  
clean_tweets_post_remove_dups = [tweet_cleaner2(clean_tweets_remove_dups2[i]) for i in range(len(clean_tweets_remove_dups2))]

177070it [29:57, 98.52it/s]


In [0]:
clean_en_interactions = pd.DataFrame(clean_tweets_inter2)
clean_en_posts = pd.DataFrame(clean_tweets_post2)

clean_inter_remove_dups = pd.DataFrame(clean_tweets_inter_remove_dups)
clean_post_remove_dups = pd.DataFrame(clean_tweets_post_remove_dups)

In [0]:
clean_en_interactions.to_csv('clean_EN_interactions.csv', index=False, header=['Post'])
clean_en_posts.to_csv('clean_EN_posts.csv', index=False, header=['Post'])

clean_inter_remove_dups.to_csv('clean_EN_interactions_remove_dups.csv', index=False, header=['Post'])
clean_post_remove_dups.to_csv('clean_EN_posts_remove_dups.csv', index=False, header=['Post'])

In [0]:
from google.colab import files
files.download('clean_EN_interactions.csv')
files.download('clean_EN_posts.csv')
files.download('clean_EN_interactions_remove_dups.csv')
files.download('clean_EN_posts_remove_dups.csv')